lapack/testlapack/dorgr2.go

```diff
@@ -0,0 +1,67 @@
+// Copyright ©2021 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package testlapack
+
+import (
+	"testing"
+
+	"golang.org/x/exp/rand"
+
+	"gonum.org/v1/gonum/floats/scalar"
+)
+
+type Dorgr2er interface {
+	Dgerqfer
+	Dorgr2(m, n, k int, a []float64, lda int, tau []float64, work []float64)
+}
+
+func Dorgr2Test(t *testing.T, impl Dorgr2er) {
+	rnd := rand.New(rand.NewSource(1))
+	for ti, test := range []struct {
+		// Dorgr2 lim: n >= m >= k >= 0
+		m, n, lda int
```

Comment:
I prefer loops to make sure that all corner-case inputs are covered including the zero sizes. Plus this lacks the variation in `k`.